### PR TITLE
ImGuiConsole now works with clipping... don't ask.

### PR DESCRIPTION
**Pull request by DerpleMQ2**
ImGuiConsole has a new property called "opacity" that you can set on creation which will multiply all theme attributes with that alpha value.

### CODE REVIEW

File: src/main/ImGuiZepEditor.cpp [hunk 1]
```diff
@@ -127,8 +127,8 @@ class ZepEditor_ImGui : public Zep::ZepDisplay, public Zep::ZepEditor
 	virtual void DrawRectFilled(const NRectf& rc, ZepColor col) const override;
 	virtual void SetClipRect(const NRectf& rc) override { m_clipRect = rc; }
 	virtual Zep::ZepFont& GetFont(ZepTextType type) override;
-
-private:
+	
+	private:
```
Comment (brainiac) [inline src/main/ImGuiZepEditor.cpp:131]: revert this diff

File: src/main/ImGuiZepEditor.cpp [hunk 2]
```diff
@@ -171,7 +171,8 @@ void ZepEditor_ImGui::DrawChars(ZepFont& font, const NVec2f& pos, ZepColor col,
 	}
 	else
 	{
-		drawList->PushClipRect(toImVec2Adjusted(m_clipRect.topLeftPx, m_screenPos), toImVec2Adjusted(m_clipRect.bottomRightPx, m_screenPos));
+		//drawList->PushClipRect(toImVec2Adjusted(m_clipRect.topLeftPx, m_screenPos), toImVec2Adjusted(m_clipRect.bottomRightPx, m_screenPos));
+		drawList->PushClipRect(toImVec2Adjusted(m_clipRect.topLeftPx, m_screenPos), toImVec2Adjusted(m_clipRect.bottomRightPx, m_screenPos), true);
```
Comment (brainiac) [inline src/main/ImGuiZepEditor.cpp:175]: remove commented out code

File: src/main/MQ2ImGuiConsole.cpp [hunk 3]
```diff
@@ -592,6 +605,7 @@ struct ImGuiZepConsole : public mq::imgui::ConsoleWidget, public mq::imgui::ImGu
 	std::shared_ptr<ZepConsoleSyntax> m_syntax;
 	int m_maxBufferLines = 10000;
 	bool m_autoScroll = true;
+	float m_alphaOverride = 1.0f;
```
Comment (brainiac) [inline src/main/MQ2ImGuiConsole.cpp:608]: unused

File: src/main/MQ2ImGuiConsole.cpp [hunk 4]
```diff
@@ -892,6 +906,13 @@ struct ImGuiZepConsole : public mq::imgui::ConsoleWidget, public mq::imgui::ImGu
 	{
 		m_maxBufferLines = maxBufferLines;
 	}
+
+	float GetOpacity() const override { return m_theme->GetOpacity(); }
+
+	void SetOpacity(float opacity) override
+	{
+		m_theme->SetOpacity(opacity);
+	}
```
Comment (brainiac) [inline src/main/MQ2ImGuiConsole.cpp:915]: these are overrides but i don't see the base class that its overriding

**Review by brainiac — CHANGES_REQUESTED** [review]

File: src/main/ImGuiZepEditor.cpp (hunk 1 above)
Comment (DerpleMQ2) [reply src/main/ImGuiZepEditor.cpp]: fixed.

File: src/main/MQ2ImGuiConsole.cpp (hunk 3 above)
Comment (DerpleMQ2) [reply src/main/MQ2ImGuiConsole.cpp]: fixed

**Pull request merged**